components/pkg-export-kubernetes/src/manifest.rs

```diff
@@ -105,3 +127,62 @@ impl Manifest {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_manifest_generation() {
+        let mut m = Manifest {
+            metadata_name: "hi-f84iq".to_owned(),
+            habitat_name: "hi".to_owned(),
+            image: "core/hi".to_owned(),
+            count: 3,
+            service_topology: Default::default(),
+            service_group: Some("group1".to_owned()),
+            config_secret_name: Some("alphabravocharlie".to_owned()),
+            ring_secret_name: Some("deltaechofoxtrot".to_owned()),
+            binds: vec![],
+        };
+
+        let expected = include_str!("../tests/KubernetesManifestTest.yaml");
+
+        let mut o = vec![];
+        m.generate(&mut o).unwrap();
+
+        let out = String::from_utf8(o).unwrap();
+
+        assert!(out == expected);
+    }
+
+    #[test]
+    fn test_manifest_generation_binds() {
+        let mut m = Manifest {
+            metadata_name: "hi-f84iq".to_owned(),
+            habitat_name: "hi".to_owned(),
```

Comment:
in the manifest template:

```
metadata:
  ## Name of the Habitat service.
  name: {{metadata_name}}                                                                                                              
  labels:
    habitat-name: {{habitat_name}}
```

IIRC it used to be just one name here but then Michal split into two names and added a random string to the former so that there can be multiple instances of a service.